internal/datasource/common/pluginconfiguration/plugin_configuration_schema.go

```diff
@@ -0,0 +1,143 @@
+package pluginconfiguration
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/listdefault"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func ToDataSourceSchema() schema.SingleNestedAttribute {
```

Comment:
I wonder if the framework has any way to convert from resource to data source schema... this is an unpleasant amount of repetition

Reply:
It doesn't seem like the framework has any solutions for this, but we could try doing through automation with the python glob package potentially. The only thing is, there are some data source schema's that do not mirror their counterpart resources very well.

Reply:
Yeah that's a good point. I guess this is what we have to do for now.